### DISCUSSION TITLE
Custom element doesn't exist: floorplan-card.

### DISCUSSION THREAD

**Discussion by ketchesonm** [ExperienceLovelace/ha-floorplan#63] [Q&A] [answered]
Hi, 
I'm just getting started with setting up the floorplan card and I've hit a snag. I was hoping for some help troubleshooting. I've followed the instructions outlined here: https://experiencelovelace.github.io/ha-floorplan/docs/quick-start/

I see an error in the lovelace card stating "Custom element doesn't exist: floorplan-card."
<img width="497" alt="Screenshot 2021-03-04 at 10 01 49" src="https://user-images.githubusercontent.com/35229656/109946936-b2382d80-7cd0-11eb-9f30-686925f3a3d4.png">

I used HACS to install. Validated that the resources are defined (see image) 
<img width="1182" alt="Screenshot 2021-03-04 at 09 54 58" src="https://user-images.githubusercontent.com/35229656/109946359-10b0dc00-7cd0-11eb-832a-f9925b056451.png">

I can see the ha-floorplan files have been uploaded
<img width="341" alt="Screenshot 2021-03-04 at 09 58 27" src="https://user-images.githubusercontent.com/35229656/109946507-3807a900-7cd0-11eb-8125-dc6155169f0f.png">

I can see the directory of the file doesn't match the resource defined - I have tried adding another resource using the url seen in the screenshot (/community/ha-floorplan/foorplan.js)

That didn't help. Any advice on things to check or potential causes would be much appreciated!

**Comment by exetico** [comment]
Check this:
https://github.com/ExperienceLovelace/ha-floorplan/discussions/49#discussioncomment-357304 

🥳

**Update**: See comments.

**Reply by exetico** [reply]
Oh snap, you've already used `www`. Have you restarted Home Assistant?

**Reply by exetico** [reply]
Also, try and force reload with Ctrl+Shift+R a few times.

**Reply by exetico** [reply]
The path defined in HA looks fine.. HACS-resources should be listed like this:
![Screenshot_20210304-115630](https://user-images.githubusercontent.com/3549445/109953795-cd129e00-7ce0-11eb-96ec-00fb98cae89a.jpg)

Are your `config` folder actually the folder with your `configuration.yaml` file?

**Reply by ketchesonm** [reply]
Thanks for your reply! 
I'll respond to each recommendation below. 

- I have restarted a few times. 
- As you noted I have placed the files within www/ folder 
- The configuration.yaml file is within the config folder. I hadn't made any changes to the configuration.yaml after installing should I have? 

> Also, try and force reload with Ctrl+Shift+R a few times.
Where would you recommend applying this? Should I be looking at the lovelace dashboard? Would. this only be needed if I haven't restarted

Other than trying this reload I'm still open to more troubleshooting suggestions since the problem is persisting

**Reply by OzGav** [reply]
You should have added the resource to your configuration.yaml if you are using yaml mode...?

**Reply by ketchesonm** [reply]
I added it via the UI at the url: /config/lovelace/resources

**Reply by exetico** [reply]
@OzGav Good point!

Else, try and see if you can open the ressource with something like:
`{HomeAssistantIP}/local/community/ha-floorplan/floorplan.js`

**Reply by exetico** [reply]
@ketchesonm - In YAML mode you will need a file for that :-)

https://www.home-assistant.io/lovelace/dashboards-and-views#adding-more-dashboards-with-yaml

Look for `resources`

**Reply by ketchesonm** [reply]
It looks like my dashboard isn't in YAML mode but UI configuration. 
<img width="1168" alt="Screenshot 2021-03-04 at 11 24 40" src="https://user-images.githubusercontent.com/35229656/109957018-45c32b80-7cdc-11eb-9eba-084e3260bf43.png">

I am able to navigate to the .js via the URL

<img width="892" alt="Screenshot 2021-03-04 at 11 21 51" src="https://user-images.githubusercontent.com/35229656/109956798-fc72dc00-7cdb-11eb-83a6-c13ef1a4d92c.png">

**Reply by ketchesonm** [reply]
Does it need to be YAML mode to work? if so, that quick start guide is misleading since it suggests adding resources via the UI.

**Reply by exetico** [reply]
And you have used the Ctrl+Shift+R a few times, and used this button, too?

![Screenshot_20210304-122558](https://user-images.githubusercontent.com/3549445/109957193-e61d4e00-7ce4-11eb-9921-b025693a17ef.jpg)

**Reply by OzGav** [reply]
No you don’t need to be in YAML mode. Just helps us to know which method you are using

**Reply by ketchesonm** [reply]
Also - Thanks to all of you for the help! I had hit a wall :P I'll  add it to my configuration.yaml now to check that this works. Just would be goo dto understand why in UI mode it wasn't working.

**Reply by exetico** [reply]
Check this for UI mode:
https://community.home-assistant.io/t/floorplan-now-available-as-a-lovelace-card/115489/323?u=exetico 

Let us know how it goes 👍.

**Reply by ketchesonm** [reply]
Oh I had given that guide a follow before as well! really useful but hadn't found the issue. 

A TWIST!!! I just looked on my home assistant app and the card renders in the app but not in my browser.. I did clear my cache.. any ideas as to why it may render differently between devices / browsers? (I'm using chrome)

**Reply by ketchesonm** [reply]
I restarted my browser and it worked - welp ... thats embarassing.. Thank you all for your help along the way!!

**Reply by exetico** [reply]
@ketchesonm So, you didn't use the button I targeted, and Ctrl+Shift+R a few times? 😁 That will secure a fresh load 😊

But it's great that it's now fixed 🎉

**Comment by exetico** [comment]
Hi @ketchesonm 

I've moved this to the Q&A category. And to wrap things up, I would like to have the actually solution for your problem.

You ended up moving to the YAML-mode, and added the resource manually, and after force reload/clear cache, everything worked as expected, right? :-) 

_If so, I'll mark this discussion as "solved"._

**Comment by ketchesonm** [comment, marked as answer]
Hi @exetico,  I stayed in the UI mode but did do a reload of the resources and then closed my browser and reopened. Then it worked!

**Reply by exetico** [reply]
Great 👍!
